### DISCUSSION TITLE
Chat example report `Publish error: Publish error: InsufficientPeers` with peers already discovered

### DISCUSSION THREAD

**Discussion by EMUNES** [libp2p/rust-libp2p#4837] [Q&A] [answered]
Hi all. I'm trying to learn about libp2p by the chat example. I have running the example in two terminals, they can find each other but they will report `Publish error` like below:

```
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
mDNS discovered a new peer: 12D3KooWCyV65nPqHkRGEG3j4eTbbCvJM5hdVADMZ3Mwavu2RgCk
df
Publish error: InsufficientPeers
sd
Publish error: InsufficientPeers
asd
Publish error: InsufficientPeers
```

I'm confused about why they can discover each other but still shows `InsufficientPeers`. Anyone has clues?

I'm running on Windows 11, `libp2p 0.53`.

**Comment by mxinden** [comment]
Do all of these peers join the same topic? Are they added to each others mesh?

If you want to dig deeper:

https://github.com/libp2p/rust-libp2p/blob/7f4ba690e87a867403f6266d8ee7d7db5e7a15bc/protocols/gossipsub/src/behaviour.rs#L603-L763

**Comment by EMUNES** [comment]
Thank you, after making sure the topic and mesh are working in the example (it is mostly copied from the examples folder). I try to play with the code and find the reason it won't work: I have removed the tracing code:

```
    let _ = tracing_subscriber::fmt()
        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
        .try_init();
```

If I add those code, the chat example works as expected. 

I'm puzzled why would tracing could actually affect how gossipsub works. I assume it should works fine even if I don't invoke tracing but it seems critical in this case. It is designed to must invoke tracing with libp2p?

**Comment by cyrus937** [comment]
I have the same problem.
I haven't changed the chat example, but sending messages doesn't work. Also, there's a latency in discovering peers.
On the first run, everything worked fine, but since then I've had this **InsufficientPeers** error

**Comment by thomaseizinger** [comment, marked as answer]
Confirmed that the example is currently broken. Fix in https://github.com/libp2p/rust-libp2p/pull/4840. Sorry about that folks!
